sdk/core/azure-core/src/credentials/credentials.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <credentials/credentials.hpp>
+#include <internal/credentials_internal.hpp>
+
+using namespace azure::core::credentials;
+
+Credential::Credential() = default;
+Credential::~Credential() noexcept = default;
+Credential::Credential(Credential const&) = default;
+Credential& Credential::operator=(Credential const&) = default;
+void Credential::SetScopes(std::string const&) {}
+
+std::shared_ptr<TokenCredential::Token> const TokenCredential::Token::Empty
+    = std::shared_ptr<TokenCredential::Token>(new Token());
```

Comment:
If the shared_ptr is kept this should use `make_shared` to save a memory allocation. Or make the class act as if it is the empty value when the shared_ptr is nullptr.

Reply:
Ditto `make_shared` everywhere else in this file.

Reply:
I didn't want the nullptr to act as empty value, because that would require me to check for null in several places, keep them updated, and also that check is going to be executed every time.

Reply:
I removed the shared_ptr, and using std::mutex now.